s4e-web/src/app/views/settings/settings.routes.ts

```diff
@@ -1,88 +1,166 @@
+              url: 'dashboard'
+            }
+          ] as IBreadcrumb[]
+        }
       },
       {
         matcher: managerDashboardMatcher,
-        component: DashboardComponent
+        component: DashboardComponent,
+        data: {
+          isAdminDashboard: false,
+          breadcrumbs: [
+            {
+              label: 'Tablica menedżera',
+              url: 'dashboard'
+            }
+          ] as IBreadcrumb[]
+        }
       },
+
       {
-        path: 'add-institution',
-        component: InstitutionFormComponent,
+        path: 'institutions',
+        component: InstitutionListComponent,
         canActivate: [IsManagerGuard],
         data: {
-          isEditMode: false
+          breadcrumbs: [
+            {
+              label: 'Instytucje'
+            }
+          ] as IBreadcrumb[]
         }
       },
       {
-        path: 'edit-institution',
-        component: InstitutionFormComponent,
+        path: 'institution',
+        component: InstitutionProfileComponent,
         canActivate: [IsManagerGuard],
         data: {
-          isEditMode: true
+          breadcrumbs: [
+            {
+              label: 'Profil instytucji'
+            }
+          ] as IBreadcrumb[]
         }
       },
+
       {
-        path: 'profile',
-        component: ProfileComponent
+        path: 'add-institution',
+        component: InstitutionFormComponent,
+        canActivate: [IsManagerGuard],
+        data: {
+          isEditMode: false,
+          breadcrumbs: [
+            {
+              label: 'Lista instytucji',
+              url: 'institutions'
+            },
+            {
+              label: 'Dodaj instytucję'
+            }
+          ] as IBreadcrumb[]
+        }
       },
       {
-        path: 'change-password',
-        component: ChangePasswordComponent
+        path: 'edit-institution',
+        component: InstitutionFormComponent,
+        canActivate: [IsManagerGuard],
+        data: {
+          isEditMode: true,
+          breadcrumbs: [
+            {
+              label: 'Profil instytucji',
+              url: 'institution'
+            },
+            {
+              label: 'Edytuj instytucję',
+            }
+          ] as IBreadcrumb[]
+        }
       },
+
       {
         path: 'groups',
         component: GroupListComponent,
-        canActivate: [IsManagerGuard]
-      },
-      {
-        path: 'groups/:groupSlug',
-        component: GroupFormComponent,
-        canActivate: [IsManagerGuard]
-      },
-      {
-        path: 'institution',
-        component: InstitutionProfileComponent,
-        canActivate: [IsManagerGuard]
-      },
-      {
-        path: 'institutions',
-        component: InstitutionListComponent,
-        canActivate: [IsManagerGuard]
+        canActivate: [IsManagerGuard],
+        data: {
+          breadcrumbs: [
+            {
+              label: 'Grupy'
+            }
+          ] as IBreadcrumb[]
```

Comment:
I'm not 100% sure it will work (the AOT compiler can be somewhat tricky with these things)

But I would suggest making a helper function - something like this:

```
export function makeBreadcrumbs(...breadcrumbs: (IBreadcrumb|string)[]): IBreadcrumb[] {
  return breadcrumbs.map(breadcrumb => typeof breadcrumb === 'string' ? {label: breadcrumb} : breadcrumb);
}
```

Then you can define breadcrumbs simply by:

```
breadcrumbs: makeBreadcrumbs('Grupy')
```